src/components/MoveAdder.ts

```diff
@@ -15,4 +15,247 @@ export class MoveAdder<TGameStartr extends FullScreenPokemon> extends GeneralCom
     public addMove(pokemon: IPokemon, move: IMove, index: number) {
         pokemon.moves[index] = move;
     }
+
+    /**
+     * Brings up the dialog for teaching a Pokemon a new move.
+     *
+     * @param pokemon   The pokemon whose moveset is being modified.
+     * @param move   The move that's going to be added into the moveset.
+     */
+    public startDialog(pokemon: IPokemon, move: IMove): void {
+        this.gameStarter.menuGrapher.deleteMenu("Yes/No");
+        this.gameStarter.menuGrapher.deleteMenu("GeneralText");
+
+        let counter = false;
+        for (const element of pokemon.moves) {
+            if (element.title.toUpperCase() === move.title.toUpperCase()) {
+                counter = true;
+            }
+        }
+        this.gameStarter.menuGrapher.createMenu("GeneralText");
+        if (counter) { //dialog for when a pokemon already knows the move
```

Comment:
Remove these `//` comments; the names of the private functions should be description enough.

If you feel the function names aren't descriptive enough, it's better to make them more descriptive than to add standalone `//` comments.